Patch test4 devices to be multi module compatible

sidey79 opened the pull request:
* **Please check if the PR fulfills these requirements**
<!-- Please do not remove these checkboxes. Check them if you have done the action behind this point -->
- [ ] Tests for the changes have been added / modified (needed for for bug fixes / features)
- [ ] commandref has been added / updated (needed for bug fixes / features)
- [ ] CHANGED has been updated (needed for bug fixes / features)


<!--Please specify if this is a bugfix, feature or update of docs and remove the other options -->

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)



* **What is the current behavior?** (You can also link to an open issue here)


* **What is the new behavior (if this is a feature change)?**



* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)



* **Other information**:

sidey79 committed:
UnitTest/tests/test_module_devices-definition.txt

Test modified to load Testdata from a JSON file and run it
modified test to work with json from SIGNALDUINO_TOOL

sidey79 commented:
@HomeAutoUser 

Hier mal das Beispiel, welches ich auf Basis deines Test entwickelt habe.

Der Code läuft auf Basis dieser JSON Datei (Exakt diese wird geladen).

https://jsoneditoronline.org/?id=b30e595a040e477eaed01f49e038bc40

Alternativ kann man auch eine URL angeben, von der eine JSON Datei geladen wird.

Wir könnten hier z.B. pro Modul z.B. eine JSON Datei mit den Testdaten hinterlegen.
Wenn Du an was anderes dachtest, dann lass mich an deinen Überlegungen teilhaben

HomeAutoUser commented:
Bitte nicht pro Modul eine JSON.
Lass uns nur einen Bestand pflegen.
Wenn dann diese https://github.com/RFD-FHEM/SIGNALduino_TOOL/blob/pre-release/FHEM/lib/SD_Device_ProtocolList.json welche ich erweitern würde mit dem Wert Modul der jeweiligen Nachricht .

sidey79 commented:
@HomeAutoUser 

Ok ich habe es mal umgebaut.
In der JSON Datei muss dafür das Modul angegeben werden, oder wir müssten das anderweitig erkennen, welches Modul gerade getestet wird. 
Ich habe das mal an drei Datensätzen beispielhaft vorgenommen.

Bei ein paar Dingen bin ich noch unsicher

1) Sollen wir wirklich das CUL_TCM_97001 Modul testen? (Genau da schlägt aktuell einer fehl)
2) Der Test wird sehr lange dauern. Das betroffene Modul zu finden wird aufwändig, da auch nicht nach Modul sortiert wird. Besser wäre, wenn wir dem Testgerüst mitgeben könnten, zu welchem Modul er die Tests laufen lassen soll.

Je Modul bräuchten wir dann einen eigenen Test, aber dafür würde es auch über Repos hinweg funktionieren.

sidey79 committed:
tests/test_module_devices-definition.txt

calulate plan and ignore errors on IT and CUL_TCM97001 module

test_SD_Protocols-definition.txt

Make it compatible with test2 suite

cpanfile

sidey79 committed:
update.yml

Checkout hole repo to get the correct timestamp

test_module_devices-definition.txt

fix url to new branch

HomeAutoUser commented:
@sidey79 
>In der JSON Datei muss dafür das Modul angegeben werden, oder wir müssten das anderweitig 

Ist erledigt und sofort bei allen unserer getesteten Daten hinterlegt.

>Ich habe das mal an drei Datensätzen beispielhaft vorgenommen.
>Bei ein paar Dingen bin ich noch unsicher
>1) Sollen wir wirklich das CUL_TCM_97001 Modul testen? (Genau da schlägt aktuell einer fehl)
>2) Der Test wird sehr lange dauern. Das betroffene Modul zu finden wird aufwändig, da auch nicht >nach Modul sortiert wird. Besser wäre, wenn wir dem Testgerüst mitgeben könnten, zu welchem >Modul er die Tests laufen lassen soll.
>
>Je Modul bräuchten wir dann einen eigenen Test, aber dafür würde es auch über Repos hinweg funktionieren.

1) vorerst würde ich erstmal nur unsere Module sicherstellen das wir eine "reine Weste" festigen.
1.1) Wenn 1 abgeschlossen, so können wir Schrittweise auch mal umfangreiche Module versuchen und ggf. nach dem Fehler suchen
2) Ist es nicht machbar bzw. sogar clever wenn wir eine Testdatei pro modul anlegen. Diese hat zwar zum großteil immer den selben Code aber so könnte man verankern, wenn die Datei `test_SD_UT_devices` zum Beispiel heißt, so muss der Test aus seinem eigenen Namen das passende Modul laden. Somit können wir pro Modul den Test erweitern oder auch jeweils spezifizieren.

Sollte das mit dem Namen auslesen aus der Testdatei funktionieren, so würde mir eine "Übersetzungsdatei" einfallen wo wir auch schnell den Testumfang der Module sehen.

EDIT: Ist hier der klick noch notwendig auf` Rebase und merge`?

sidey79 commented:
> 1. vorerst würde ich erstmal nur unsere Module sicherstellen das wir eine "reine Weste" festigen.

Sehe ich auch so.

> 2. Ist es nicht machbar bzw. sogar clever wenn wir eine Testdatei pro modul anlegen. Diese hat zwar zum großteil immer den selben Code aber so könnte man verankern, wenn die Datei `test_SD_UT_devices` zum Beispiel heißt, so muss der Test aus seinem eigenen Namen das passende Modul laden. Somit können wir pro Modul den Test erweitern oder auch jeweils spezifizieren.

Könnte man machen, aber dann hat man dauert doppelten code.
 
> Sollte das mit dem Namen auslesen aus der Testdatei funktionieren, so würde mir eine "Übersetzungsdatei" einfallen wo wir auch schnell den Testumfang der Module sehen.

Was stellst Du dir da vor? Ich kann dir leider nicht folgen.

> EDIT: Ist hier der klick noch notwendig auf` Rebase und merge`?

Ein normaler merge tut es auch.


Mit Anpassung der JSON schlagen jetzt noch einige weitere Sensoren fehl

HomeAutoUser commented:
>Mit Anpassung der JSON schlagen jetzt noch einige weitere Sensoren fehl

Kann ich das Ergebnis schon irgendwo sehen?

>Was stellst Du dir da vor? Ich kann dir leider nicht folgen.

sofort ;-)

PS: Merge, durchgeführt
